source/fundamentals/data-formats/document-data-format-record.txt

```diff
@@ -0,0 +1,101 @@
+.. _fundamentals-records:
+
+=============================
+Document Data Format: Records
+=============================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to store and retrieve data in the {+driver-long+}
+using **Java records**. Java records are a type of Java class often used to
+model data and separate business logic from data representation.
+
+.. tip::
+
+   You can declare Java records in Java 14 or later. Learn more about the
+   functionality and restrictions of records from `Java 14 Language Updates: Records <https://docs.oracle.com/en/java/javase/14/language/records.html>`__.
+
+   If you are using an earlier version of Java, you can use plain old Java
+   objects instead. See the :ref:`<fundamentals-pojos>` guide for
+   implementation details.
+
+In this guide, you can learn how to perform the following tasks:
+
+- Write data from records into MongoDB.
+- Retrieve documents from MongoDB as records.
+
+.. _fundamentals-example-record:
+
+Serialize and Deserialize a Record
```

Comment:
As with POJOs, record components can be annotated with:

- `org.bson.codecs.pojo.annotations.BsonId`
- `org.bson.codecs.pojo.annotations.BsonProperty`
- `org.bson.codecs.pojo.annotations.BsonRepresentation`

We should mention this, and perhaps show an example of their use.

Reply:
I was planning to add information on this for the next ticket, https://jira.mongodb.org/browse/DOCSP-22977
Does that work for you?